### DISCUSSION TITLE
Usage with meson?

### DISCUSSION THREAD

**Discussion by MTRNord** [SOCI/soci#1100] [Q&A]
Hi :)

I thought it was straightforward due to the cmake compat of meson but I am stuck at `../meson.build:9:12: ERROR: Dependency "soci_core" not found, tried pkgconfig and cmake`.

I am doing `soci_core = dependency('soci_core')` basically.

I tried also `soci`, `Soci`, `SOCI`, `SOCI::soci_core`, `libsoci`, `libsoci_core` and read https://soci.sourceforge.net/doc/release/4.0/installation/#using-library

But I didnt get it working. I am on fedora 38 and have `soci-devel` installed. So I am kinda stuck :)

**Comment by MTRNord** [comment]
I am not sure if this is _correct_ but this worked in the end:

SOCI.wrap:
```
[wrap-file]
directory = soci-4.0.3
source_url = https://github.com/SOCI/soci/archive/refs/tags/v4.0.3.tar.gz
source_filename = soci-4.0.3.tar.gz
source_hash = 4b1ff9c8545c5d802fbe06ee6cd2886630e5c03bf740e269bb625b45cf934928
```

meson.build snippet:
```
cmake = import('cmake')
opt_var = cmake.subproject_options()

# Globally override the C++ standard to c++20
opt_var.set_override_option('cpp_std', 'c++20')
opt_var.set_override_option('cpp_std', 'c++20', target: 'soci_core')
opt_var.set_override_option('cpp_std', 'c++20', target: 'soci_postgresql')
opt_var.add_cmake_defines({'WITH_BOOST': false, 'SOCI_ORACLE': false, 'SOCI_ODBC': false, 'SOCI_FIREBIRD': false, 'SOCI_STATIC': false, 'CMAKE_CXX_STANDARD': '20'})

soci = cmake.subproject('SOCI', options: opt_var)
soci_core = soci.dependency('soci_core')
soci_postgresql = soci.dependency('soci_postgresql')
```

**Comment by Lihis** [comment]
Since this https://bodhi.fedoraproject.org/updates/FEDORA-2026-a9d98b8068 update has landed to Fedora F43 stable repository this now works:

```
dependency('SOCI', method: 'cmake', components: ['MySQL', 'SQLite3'])
```

when you have `soci-mysql-devel` and `soci-sqlite3-devel` packages installed. Or if have installed all the `soci-<backend>-devel` packages you can drop the `components` part.

EDIT: or install backends you like and adapt `components` respectively.
